components/vni.js

```diff
@@ -0,0 +1,63 @@
+// Vni.js
+
+var _ = require('underscore');
+var Lm = require('./Lm');
+var state = require('./state');
+
+// Hash of vnis
+vnis = []; 
+
+// Define Vni class and its subclasses - InterNodeVni & IipVni
+function Vni( vnid ) {
+  this.vnid = vnid;
+  this.lm = Lm();
+  this.allInputStates = state.allInputStates;
```

Comment:
I am uncomfortable having member functions stored in VNIs and States, because they consume memory and there may be many VNI and State objects when scaling up.  For encapsulation, functions like allOutputStates can be in this VNI module (kind of like static methods), but I don't see any real need to store them in the VNI objects themselves.  I would rather keep the VNI and State objects as lean as possible.